uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChangeTest.php

```diff
@@ -1038,5 +1039,25 @@ private function deleteDefaultWorkPattern() {
     $workPatternTable = WorkPattern::getTableName();
     CRM_Core_DAO::executeQuery("DELETE FROM {$workPatternTable} WHERE is_default = 1");
   }
+
+  public function testGetBalanceChangeForToilLeaveRequest() {
+    $absenceType = AbsenceTypeFabricator::fabricate([
+      'allow_accruals_request' => true,
+    ]);
```

Comment:
The saveBalanceChange method on the TOIL request BAO calls the calculateToilExpiryDate method of the AbsenceType BAO. This will throw an exception if the absence type does not allow accrual request.